### PR TITLE
[Fix](Variant) fix variant schema change may cause invalid block schema and write missing blocks

### PR DESCRIPTION
**Pull request by eldenmoon**
…ma and write missing blocks

1. update_rowset_schema should not update _context.tablet_schema, since it's used as src schema for schema change, if it's changed may lead to wrong schema with block
2. buffering block during schema change should merge first and then flush, otherwise  may lost current block

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/rowset/segment_creator.cpp [hunk 1]
```diff
@@ -315,14 +315,15 @@ Status SegmentCreator::add_block(const vectorized::Block* block) {
     size_t block_row_num = block->rows();
     size_t row_avg_size_in_bytes = std::max((size_t)1, block_size_in_bytes / block_row_num);
     size_t row_offset = 0;
-
     if (_segment_flusher.need_buffering()) {
+        RETURN_IF_ERROR(_buffer_block.merge(*block));
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_creator.cpp:319]: need more discussion

**Review by xiaokang — COMMENTED** [review]
pls add testcase

**Comment by eldenmoon** [comment]
> pls add testcase

done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Pull request merged**